### ISSUE TITLE
Is there an option to specify where an inlined script would be placed?

### ISSUE THREAD

**Issue by Birowsky** [numical/script-ext-html-webpack-plugin#6]
Something like: body end, body beginning, head end, head beginning.

Or is the ordering not part of this plugin's scope?

**Comment by numical** [comment]
This plugin should respect [HtmlWebpackPlugin's `inject` configuration](https://github.com/ampedandwired/html-webpack-plugin#configuration).
Not quite as flexible as you are after, but it's a start...

**Comment by Birowsky** [comment]
Not quite my tempo :} 

I need to be able to specify the order of scripts. But in the `inject` configuration, I'm only able to specify where all scripts are going to be placed. Nonetheless, this seems to not be in the scope of this plugin. I'll just keep looking.

**Comment by sidonaldson** [comment]
I would like to also add weight to this request. It's very common to want to inline a script in the head of the page to intercept lack of features before the page renders. Is there any way an option would be added like `inlineHead` or similar?

**Comment by numical** [comment]
So [issue 15](https://github.com/numical/style-ext-html-webpack-plugin/issues/15) has been raised covering similar ground.  So there seems to be some interest in this functionality (at least by the standards of this plugin!).
Hence I'm looking at extending the plugin to work even when HtmlWepbackPlugin's `inject` option is `false` __and__ to offer a `position` option as per @Birowsky's request...
I'll keep you posted (and/or, if you're really keen, you cn keep an eye on the ['inject-false' branch](https://github.com/numical/style-ext-html-webpack-plugin/tree/inject-false-option).

**Comment by sidonaldson** [comment]
@numical excellent news! It would certainly fill a feature gap as I can't find an eloquent way to manage included yet hoist one or two to the <head/>

**Comment by numical** [comment]
Okey dokey - release v3.2.0 no longer requires a truthy `inject` value, and also allows [positioning of style elements](https://github.com/numical/style-ext-html-webpack-plugin#use-case-specifying-position-of-style-element).  Let me now what you think, if you have any issues etc.

**Comment by numical** [comment]
OK, I have managed [something similar for styles](https://github.com/numical/style-ext-html-webpack-plugin#use-case-specifying-position-of-style-element) in the sister plugin [style-ext-webpack-plugin](https://github.com/numical/style-ext-html-webpack-plugin) so now to look into this for scripts...

**Comment by jantimon** [comment]
@numical this would also work with inject:true

**Comment by numical** [comment]
Yup - the style-ext plugin's implementation of explicit  positioning simply overwrites the HTML so works whatever html-webpack-plugin's `inject`value is. I dont say as much in the README as I thought it might confuse. For script-ext, I wanted to do something more elegant and you have given me the clue I need! Thanks!

**Comment by numical** [comment]
@jantimon - advice please.  I  now have the [body and head arrays sorted](https://github.com/numical/script-ext-html-webpack-plugin/blob/explicit-position/lib/sort.js#L34) on a branch.  However as I use your `..alter-assets..` event, these sorted arrays are [simply appended](https://github.com/jantimon/html-webpack-plugin/blob/master/index.js#L531) to the body and head elements within your `injectAssetsIntoHtml` function.
To enable head top/bottom &body top/bottom positioning I have three options:
* submit a PR for html-webpack-plugin with body-top, body-bottom, head-top and head-bottom arrays;
* revert this plugin to work like the style-ext plugin, overwriting your html output in the `...after-html-processing...` event;
* another brilliant solution I have not thought of yet.

Your preference / recommendation?

**Comment by sidonaldson** [comment]
@numical a solution which doesn't depend on a PR for html-webpack-plugin would be ideal i think :)

**Comment by numical** [comment]
But riskier and more prone to errors. Easier to manipulate JavaScript constructs representing link tags than to manipulate generated HTML via easy-to-screw-up regex's.

**Comment by sidonaldson** [comment]
@numical Yes, i see your point. Could this only work if `inject:false` for html-webpack-plugin?

**Comment by numical** [comment]
Actually the regex option could (theoretically) work with either - though I need to check how much of html-webpack-plugin's functionality I would be reproducing in script-ext if it were to work with `inject:false`.
I must admit to being a little concerned as to whether I am opening a can of worms with this.  The next request might be to position some scripts and not others - which will fall foul of the plugin's current tag-based configuration options.

**Comment by numical** [comment]
Further commentary, same [issue](https://github.com/numical/style-ext-html-webpack-plugin/issues/16) in sibling plugin [style-ext-ebpack-plugin](https://github.com/numical/style-ext-html-webpack-plugin).

**tl;dr** : this will not go into core `html-webpack-plugin` functionality so will look at this given time.  PR's welcome!

**Comment by numical** [comment]
Have not found time for this in 9 months so I think we can assume this will not happen.
Unless someone has time for a (tested) PR...
